src/stable/build/commands/compile/get_context.ts

```diff
@@ -65,3 +84,62 @@ function getEnvVars(canisterConfig: CanisterConfig): EnvVars {
             return [envVarName, envVarValue];
         });
 }
+
+/**
+ * Finds the project root directory through multiple fallback methods.
+ * @returns The determined project root directory path
+ *
+ * @remarks
+ * First tries using npm_config_prefix (if script called with --prefix),
+ * then attempts to use npm_package_json environment variable,
+ * then searches for package.json in parent directories,
+ * and finally falls back to the current working directory if all else fails.
+ */
+async function findProjectRoot(): Promise<string> {
+    // Method 1: Check if the script was called with an explicit prefix
+    if (process.env.npm_config_prefix !== undefined) {
+        const prefixPath = process.env.npm_config_prefix;
+
+        const projectRoot = findNearestProjectRoot(prefixPath);
+
+        if (projectRoot !== undefined) {
+            return projectRoot;
+        }
+    }
+
+    // Method 2: Try to use npm_package_json which points to package.json
+    if (process.env.npm_package_json !== undefined) {
+        if (existsSync(process.env.npm_package_json)) {
+            return dirname(process.env.npm_package_json);
+        }
+    }
+
+    // Method 3: Check for presence of expected files/directories in ancestor directories
+    // Look for package.json in parent directories
+    const projectRoot = findNearestProjectRoot(process.cwd());
+
+    if (projectRoot !== undefined) {
+        return projectRoot;
+    }
+
+    // Fall back to current working directory
+    return process.cwd();
```

Comment:
Consider if this should be an error case

Reply:
Done

Reply:
Done